tests/network/wireguard.pm

```diff
@@ -58,6 +60,21 @@ sub run {
         $remote     = '10.0.2.101';
     }
 
+    if (is_sle()) {
+        add_suseconnect_product('sle-module-desktop-applications');
+        script_run('source /etc/os-release');
+        # Workaround https://bugzilla.suse.com/show_bug.cgi?id=1181941
+        my $scn = script_run('SUSEConnect -p sle-we/${VERSION_ID}/${CPU} -r ' . get_required_var('SCC_REGCODE_WE'));
+        die if ($scn ne 0 && $scn ne 106);
+        zypper_call '--gpg-auto-import-keys ref';
+        # Workaround https://bugzilla.suse.com/show_bug.cgi?id=1182004
+        $scn = script_run('SUSEConnect -p PackageHub/${VERSION_ID}/${CPU}');
+        die if ($scn ne 0 && $scn ne 106);
+        zypper_call '--gpg-auto-import-keys ref';
```

Comment:
shouldn't add_suseconnect_product take care of both cases?

Reply:
It doesn't. That's what I needed to find out today step by step. At least if there is no undocumented option that allows this.

Reply:
```perl
add_suseconnect_product('PackageHub', undef, undef, undef, 300, 1)
```

Reply:
According to the code this shouldn't help:

```perl
    my $result = script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
    if ($result != 0 && $retry) {
```

This retry option would help to workaround https://bugzilla.suse.com/show_bug.cgi?id=1124318 (which got resolved in the meantime) but the problem now is that SUSEConnect doesn't run into an error but also doesn't import the repo GPG key: https://bugzilla.suse.com/show_bug.cgi?id=1182004

I think I can replace the PackageHub `script_run` call with `add_suseconnect_product` but I will need to keep the `zypper_ref` call to import the gpg key of the PackageHub repo.

Reply:
The WE module actually seems to still need the retry option - and also the `zypper --gpg-auto-import-keys ref` as well.

Reply:
Updated PR and verification run.